CMakeLists.txt

```diff
@@ -0,0 +1,1132 @@
+cmake_minimum_required(VERSION 3.15.1)
+
+project(wasi-libc
+  LANGUAGES ASM C)
+
+include(CMakeDependentOption)
+
+option(ENABLE_THREADS "enable threads" OFF)
+option(BUILD_DLMALLOC "build dlmalloc" YES)
+cmake_dependent_option(BUILD_LIBC_BOTTOM_HALF "build libc bottom half" YES
+  "BUILD_DLMALLOC" YES)
+cmake_dependent_option(BUILD_LIBC_TOP_HALF "build libc top half" YES
+  "BUILD_LIBC_BOTTOM_HALF" YES)
+
+if(CMAKE_C_COMPILER_ID MATCHES Clang)
+  add_compile_options(-target wasm32-unknown-wasi)
```

Comment:
We generally abbreviate the target as "wasm32-wasi".

Reply:
Ah, internally, LLVM will actually normalize the triple to this.  Is there a reason to not just use the normalized form?  If its just to maintain uniformity with Make (which I think longer term could be replaced with the metabuild), that is reasonable.